### DISCUSSION TITLE
service not starting on boot

### DISCUSSION THREAD

**Discussion by SadGamerGeek** [Koenkk/zigbee2mqtt#8238] [General]
I've had z2m running well on a pi3 (clean OS install) for a couple of weeks, but just had a power outage and noticed it didn't come up when the pi did.

I'd followed the guide and my service file looks like this:

```
[Unit]
Description=zigbee2mqtt
After=network.target

[Service]
ExecStart=/usr/bin/npm start
WorkingDirectory=/opt/zigbee2mqtt
StandardOutput=inherit
# Or use StandardOutput=null if you don't want Zigbee2MQTT messages filling syslog, for more options see systemd.exec(5)
StandardError=inherit
Restart=always
User=pi

[Install]
WantedBy=multi-user.target
```

After boot, the service looks like this:

```
root@raspberrypi:~# service zigbee2mqtt status
● zigbee2mqtt.service - zigbee2mqtt
   Loaded: loaded (/etc/systemd/system/zigbee2mqtt.service; disabled; vendor preset: enabled)
   Active: inactive (dead)
```

I noticed /opt/zigbee2mqtt was owned by root, so tried the user parameter as root instead in the service file. That didn't work, so I put it back to pi, and then did a `chown -R pi /opt/zigbee2mqtt/` to change it to match the user in the service file, but that didn't help either.

I'm struggling to see any useful logging info, but the end of the last log seems to show z2m being shut down cleanly:

```
info  2021-08-03 13:32:29: MQTT publish: topic 'zigbee2mqtt/study rhs', payload '{"brightness":169,"color":{"x":0.3131,"y":0.3232},"color_mode":"color_temp","color_temp":153,"last_seen":"2021-08-03T13:32:29+01:00","linkquality":132,"state":"OFF","update":{"state":"idle"}}'
info  2021-08-03 13:33:25: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'offline'
info  2021-08-03 13:33:25: Disconnecting from MQTT server
info  2021-08-03 13:33:25: Stopping zigbee-herdsman...
info  2021-08-03 13:33:36: Stopped zigbee-herdsman

```

service start zigbee2mqtt works as expected from both users pi (where it asks for the password) and root.

What might I be missing?

**Comment by alexhall1986** [comment]
hi did you resolve this, I am having same issue

**Reply by SadGamerGeek** [reply]
I'm afraid I didn't resolve it as a standalone install, but I now run it as a Home Assistant add-in, and it now starts OK.
